spatialos-sdk/src/worker/snapshot.rs

```diff
@@ -18,38 +18,36 @@ impl SnapshotOutputStream {
             component_vtables: DATABASE.to_worker_sdk(),
             default_component_vtable: std::ptr::null(),
         };
+        let stream_ptr =
+            unsafe { Worker_SnapshotOutputStream_Create(filename_cstr.as_ptr(), &params) };
 
-        let ptr = unsafe { Worker_SnapshotOutputStream_Create(filename_cstr.as_ptr(), &params) };
-
-        let stream = SnapshotOutputStream { ptr };
-
-        let err_ptr = unsafe { Worker_SnapshotOutputStream_GetError(ptr) };
-        if !err_ptr.is_null() {
-            unsafe { Worker_SnapshotOutputStream_Destroy(ptr) };
-            return Err(cstr_to_string(err_ptr));
+        let state = unsafe { Worker_SnapshotOutputStream_GetState(stream_ptr) };
+        match Worker_StreamState::from(state.stream_state) {
+            Worker_StreamState_WORKER_STREAM_STATE_GOOD => {
```

Comment:
Are the error codes that this can return meaningful?

Reply:
If so - should we create an actual error enum to return here (and elsewhere)?

Reply:
Done.